### PR TITLE
Added: Linear space interpolation

### PR DESCRIPTION
**Pull request by dvdvideo1234**

### CODE REVIEW

**Comment by Kefta** [comment]
You should put this in the math library and rename it to LinearSet or something more descriptive of its single dimensional nature. Also, add some bound checks to prevent infinite loops. Lastly, you should match the codebase:  add spaces around the parentheses and put the loop body and return on separate lines.

**Comment by dvdvideo1234** [comment]
@Kefta

Well it is in the math library ;) Done rename it to `LinearSet`

**Comment by Kefta** [comment]
Yes but this code is being distributed to all clients. Although the codebase is somewhat varied throughout, there are some pretty consistent syntax rules that are followed in code additions. Ideally, files should look like it they were written by one person using a uniform style.

**Comment by Kefta** [comment]
misread whoopS

**Comment by Kefta** [comment]
It doesn't have math. in front of it so right now it's a global function.

**Comment by dvdvideo1234** [comment]
@Kefta Done, thank you!

File: garrysmod/lua/includes/extensions/math.lua [hunk 1]
```diff
@@ -217,3 +217,21 @@ end
 function math.Remap( value, inMin, inMax, outMin, outMax )
 	return outMin + ( ( ( value - inMin ) / ( inMax - inMin ) ) * ( outMax - outMin ) )
 end
+
+--[[---------------------------------------------------------
+	Name: GetLinearSpace(nStart, nEnd, nAmount)
```
Comment (CornerPin) [inline garrysmod/lua/includes/extensions/math.lua]: You forgot to change this line.

Comment (dvdvideo1234) [reply garrysmod/lua/includes/extensions/math.lua]: Done, thanks !

**Comment by notcake** [comment]
> You should put this in the math library and rename it to LinearSet or something more descriptive of its single dimensional nature.

It's called `linspace` in matlab

https://www.mathworks.com/help/matlab/ref/linspace.html

**Comment by dvdvideo1234** [comment]
> LinearSet

https://github.com/Facepunch/garrysmod/pull/1563/commits/1696b902ea77005a2b7f34898baa8156aef4e779

**Comment by thegrb93** [comment]
Why are we adding matlab functions to gmod's math library?

**Comment by dvdvideo1234** [comment]
So, is it gonna be added? This is used for interpolation mainly.

**Comment by robotboy655** [comment]
I agree with @thegrb93, this is a quite narrow use case function, we already have a linear interpolation function (https://wiki.facepunch.com/gmod/Global.Lerp, vector and angle versions also exist), you can generate a table of points easily enough with it.

```
function math.LinearSet( nStart, nEnd, nAmount )
	local t = {}
	for i=0, nAmount - 1 do
		table.insert( t, Lerp( i / (nAmount - 1), nStart, nEnd ) )
	end
	return t
end

PrintTable( math.LinearSet(0, 10, 5) ) --> {0, 2.5, 5, 7.5, 10 }
PrintTable( math.LinearSet(-5,5,7) ) --> { -5.0000, -3.3333, -1.6667, 0, 1.6667, 3.3333, 5.0000 }
```

Your implementation doesn't even match what matlab function does.

**Comment by dvdvideo1234** [comment]
Neat.. Will try ;)

**Pull request closed without merging**